### PR TITLE
Makes crafting items put the result on a rack/table in front of the user if the user's hands are full

### PR DESCRIPTION
**Pull request by CydiaLamiales**
## About The Pull Request
Makes crafting items put the result on a rack/table in front of the user if the user's hands are full

## Why It's Good For The Game
It would make sense that you'd put something on a table after making it instead of dropping it on the floor.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/BeeStation/BeeStation-Hornet/assets/110184118/34a23317-75e2-45f3-8dae-08af2822a74a

</details>

## Changelog
:cl:
tweak: Made crafting an item be put on a table or rack in front of the user if their hands are full
/:cl:

### CODE REVIEW

**Review by PowerfulBacon — COMMENTED** [review]
Make it only happen for crafted items and not structures

**Comment by CydiaLamiales** [comment]
> Make it only happen for crafted items and not structures

There's an isitem(result) check before the code, line 201

**Comment by EvilDragonfiend** [comment]
I think you can put an item behind a directional window

**Comment by CydiaLamiales** [comment]
> I think you can put an item behind a directional window

Fixed

File: code/datums/components/crafting/crafting.dm [hunk 1]
```diff
@@ -199,7 +199,15 @@
 	log_crafting(user, TR.name, result, TR.dangerous_craft)
 	if(!istext(result)) //We made an item and didn't get a fail message
 		if(ismob(user) && isitem(result)) //In case the user is actually possessing a non mob like a machine
-			user.put_in_hands(result)
+			if(!user.put_in_hands(result))
+				var/turf/front_turf = get_step(user, user.dir)
+				if(user.TurfAdjacent(front_turf))
+					var/obj/structure/rack/found_rack = locate(/obj/structure/rack) in front_turf
+					var/obj/structure/table/found_table = locate(/obj/structure/table) in front_turf
+					if(found_rack || found_table)
```
Comment (EvilDragonfiend) [inline code/datums/components/crafting/crafting.dm]:
```suggestion
					if((locate(/obj/structure/table) in front_turf) || (locate(/obj/structure/rack) in front_turf))
```

**Review by EvilDragonfiend — CHANGES_REQUESTED** [review]
variables are not used. just doing it in `if()` and having || is better because you won't calculate more unnecessary things
also finding table first is cheaper because it's more common

**Review by EvilDragonfiend — APPROVED** [review]

**Pull request merged**